src/components/DropList/DropList.utils.js

```diff
@@ -201,43 +201,36 @@ export function getEnabledItemIndex({
 }) {
   let enabledItemIndex = 0
 
-  if (arrowKey === 'UP') {
-    for (let index = items.length - 1; index >= 0; index--) {
-      const isGroupOrDividerItem = checkIfGroupOrDividerItem(items[index])
-
-      if (
-        !isGroupOrDividerItem &&
-        !items[index].isDisabled &&
-        (nextHighlightedIndex === 0 || index <= nextHighlightedIndex)
-      ) {
-        enabledItemIndex = index
-        break
-      }
-    }
-  }
+  const isNextIndexItemHighlightable =
+    !checkIfGroupOrDividerItem(items[nextHighlightedIndex]) &&
+    !items[nextHighlightedIndex].isDisabled
 
-  if (arrowKey === 'DOWN') {
-    if (currentHighlightedIndex === nextHighlightedIndex) {
-      return getEnabledItemIndex({
-        currentHighlightedIndex,
-        nextHighlightedIndex: 0,
-        items,
-        arrowKey,
-      })
+  if (
+    isNextIndexItemHighlightable &&
+    currentHighlightedIndex !== nextHighlightedIndex
+  ) {
+    enabledItemIndex = nextHighlightedIndex
```

Comment:
What about returning here early and do not add `else` on the next line. Would make the code a bit more clearer I think (at least there would be one less indentation)